examples/new_maps.py

```diff
@@ -0,0 +1,290 @@
+parser.add_option('--beta-sync-gaus', dest='beta_sync_gaus', default=False, action='store_true',
+                  help='Set to include gaussian spectral index for sync, default=False')
+parser.add_option('--sigma-d', dest='sigma_dust', default=0,  type=int,
+                  help='Select amplitude of dust variation, default=0. Input values are multiplied to E-2')
+parser.add_option('--sigma-s', dest='sigma_sync', default=0,  type=int,
+                  help='Select amplitude of sync variation, default=0. Input values are multiplied to E-2')
+
+(o, args) = parser.parse_args()
+
+nside = o.nside
+seed= o.seed
+np.random.seed(seed)
+
+# Dust
+A_dust_BB=5
+EB_dust=2.  
+alpha_dust_EE=-0.42 
+alpha_dust_BB=-0.2
+nu0_dust=353. 
+temp_dust = 19.6
+if o.beta_dust_gaus:
+    beta_dust =  hp.ud_grade(hp.read_map('map_beta_dust_sigD%d_sd%d.fits'%(o.sigma_dust, seed), verbose=False), nside_out=nside)
+else:
+    beta_dust = 1.59
+
+# Sync
+A_sync_BB=2
+EB_sync=2.
+alpha_sync_EE=-0.6
+alpha_sync_BB=-0.4
+nu0_sync=23.
+if o.beta_sync_gaus:
+    beta_sync = hp.ud_grade(hp.read_map('map_beta_sync_sigS%d_sd%d.fits'%(o.sigma_sync, seed), verbose=False), nside_out=nside)
+else:
+    beta_sync=-3.
+
+nu = np.array([27., 39., 93., 145., 225., 280.])
+nfreq = len(nu)
+
+# Define CMB spectrum
+def fcmb(nu):
+    x=0.017608676067552197*nu
+    ex=np.exp(x)
+    return ex*(x/(ex-1))**2
+
+# All spectra
+if o.use_pysm:
+    dirname = "new_simulation_ns%d_seed%d_pysm"%(nside, seed)
+    def power_law(nu, nu0, b):
+        return (nu/nu0)**b
+    
+    def B(nu, t):
+        x = constants.h*nu*1E9 / constants.k / t
+        return  (nu)**3  / np.expm1(x)
+    
+    def black_body(nu, nu0, t):
+        return B(nu, t)/B(nu0, t)
+
+    def comp_sed(nu,nu0,beta,temp,typ):
+        if typ=='cmb':
+            return fcmb(nu)
+        elif typ=='dust':
+            return power_law(nu, nu0, beta-2)*black_body(nu, nu0, temp)
+        elif typ=='sync':
+            return power_law(nu, nu0, beta)            
+else:
+    dirname = "new_simulation_ns%d_seed%d_bbsim"%(nside, seed)
+    def comp_sed(nu,nu0,beta,temp,typ):
+        if typ=='cmb':
+            return fcmb(nu)
+        elif typ=='dust':
+            x_to=0.04799244662211351*nu/temp
+            x_from=0.04799244662211351*nu0/temp
+            return (nu/nu0)**(1+beta)*(np.exp(x_from)-1)/(np.exp(x_to)-1)
+        elif typ=='sync':
+            return (nu/nu0)**beta
+        return None
+
+os.system('mkdir -p '+dirname)
+    
+if type(beta_dust)==np.ndarray:
+    for i in beta_dust:
+        f_dust = comp_sed(nu, nu0_dust, i, temp_dust, 'dust')
+else:
+    f_dust = comp_sed(nu, nu0_dust, beta_dust, temp_dust, 'dust')    
+if type(beta_sync)==np.ndarray:
+    for j in beta_sync:
+        f_sync = comp_sed(nu, nu0_sync, j, None, 'sync')
+else:
+    f_sync = comp_sed(nu, nu0_sync, beta_sync, None, 'sync')
+f_cmb = fcmb(nu)
+
+f_dust = f_dust/f_cmb
```

Comment:
I think if you get rid of all the non-pysm options, you won't need f_dust or f_sync for anything

Reply:
And actually the same for f_cmb. You only really need f_cmb_RJ